frontend/src/v5/ui/routes/dashboard/projects/drawings/drawingsList/drawingsListItem/drawingsListItemLoading.component.tsx

```diff
@@ -0,0 +1,87 @@
+/**
+ *  Copyright (C) 2024 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import {
+	DashboardListItemIcon,
+	DashboardListItemRow,
+} from '@components/dashboard/dashboardList/dashboardListItem/components';
+import { FavouriteCheckbox } from '@controls/favouriteCheckbox';
+
+import { DashboardListItem } from '@components/dashboard/dashboardList';
+import { Drawing } from '@/v5/store/drawings/drawings.types';
+import { DashboardParams } from '@/v5/ui/routes/routes.constants';
+import { DrawingsActionsDispatchers } from '@/v5/services/actionsDispatchers';
+import { SkeletonBlock } from '@controls/skeletonBlock/skeletonBlock.styles';
+import { DrawingsListItemTitle } from './drawingsListItemTitle/drawingsListItemTitle.component';
+import { FixedOrGrowContainer } from '@controls/fixedOrGrowContainer';
+import { DrawingsEllipsisMenu } from './drawingsEllipsisMenu/drawingsEllipsisMenu.component';
+import { DRAWING_LIST_COLUMN_WIDTHS } from '@/v5/store/drawings/drawings.helpers';
+
+interface IDrawingListItem {
+	drawing: Drawing;
+	delay?: number;
+}
+
+export const DrawingListItemLoading = ({
+	drawing,
+	delay,
+}: IDrawingListItem): JSX.Element => {
+	const { teamspace, project } = useParams<DashboardParams>();
+
+	const onChangeFavourite = ({ currentTarget: { checked } }) => {
```

Comment:
This doesn't work, the "add/remove favourite" only works for the item that was already favourite

https://github.com/3drepo/3drepo.io/assets/43547703/870f364e-8a6a-4509-b00c-e6f357fdfa61

> **Note:** I tweaked `DrawingsList.component:124` to look like this `drawing.hasStatsPending || true`

Reply:
Couldn't reproduce this, but might have been because of other changes I have since made